packages/patternfly-4/react-core/src/components/Pagination/Navigation.tsx

```diff
@@ -111,13 +117,27 @@ export class Navigation extends React.Component<NavigationProps, NavigationState
     }
   }
 
+  handleInputWidth = () => {
+    const { lastPage } = this.props;
+    const len = String(lastPage).length;
+    if (len >= 3) {
+      if (this.inputRef.current) {
+        this.inputRef.current.style.setProperty(inputTotalWidth.name, `calc(${inputWidthBase.value} + ${len} * 1ch)`);
```

Comment:
Is there not a way to solve this with normal CSS upstream in https://github.com/patternfly/patternfly-next? If not, resorting to CSS-in-JS is okay with me.

Reply:
This looks like a style that belongs in pf-core

Reply:
To set the input width, from the solution in this comment https://github.com/patternfly/patternfly-react/issues/3530#issuecomment-576937340, you want to use `setProperty` to define `--pf-c-pagination__nav-page-select--c-form-control--width-chars` with the number of characters you want the input to display. The `calc()` and `1ch` conversion happens in the CSS here:

https://github.com/patternfly/patternfly-next/blob/72790c83d8cc1704f831e42ba06bf828c67d029f/src/patternfly/components/Pagination/pagination.scss#L25

Reply:
I tried to update `--pf-c-pagination__nav-page-select--c-form-control--width-chars` but it didn't work maybe you can find why.

Reply:
What element are you calling `setProperty()` on? Currently it will need to be added to the element with `.pf-c-pagination` (the component's outer most container)

Reply:
@mcoker thanks - I did as you said and it works now!